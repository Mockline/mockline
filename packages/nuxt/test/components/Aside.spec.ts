import { describe, it, expect } from 'vitest'
import type {TypeOf} from "zod";
// @ts-ignore
import { MAside } from '#components'
import ComponentRender from "../component-render";

describe('Aside', () => {
  it.each([
    ['sticky', { props: { sticky: true } }],
    ['not sticky', { props: { sticky: false } }],
    ['<MAside class="text-red-500" sticky><template #default>default</template><template #bottom>bottom</template></MAside>']
    // @ts-expect-error
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof MAside.props>) => {
    const html = await ComponentRender(nameOrHtml, options, MAside)
    expect(html).toMatchSnapshot()
  })
})
