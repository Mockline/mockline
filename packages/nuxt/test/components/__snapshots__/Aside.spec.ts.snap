// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Aside > renders <MAside class="text-red-500" sticky><template #default>default</template><template #bottom>bottom</template></MAside> correctly 1`] = `
"<aside class="overflow-y-auto sticky top-[var(--header-height)] text-red-500">
  <div class="relative">defaultbottom</div>
</aside>"
`;

exports[`Aside > renders not sticky correctly 1`] = `
"<aside class="overflow-y-auto">
  <div class="relative"></div>
</aside>"
`;

exports[`Aside > renders sticky correctly 1`] = `
"<aside class="overflow-y-auto sticky top-[var(--header-height)]">
  <div class="relative"></div>
</aside>"
`;
